README.md

```diff
@@ -10,7 +10,22 @@ Model documentation in Markdown can also be generated for this repository, and i
 
 ### Automated generation of YAML
 
-The model YAML files are programmatically generated from [the Google Sheet](https://docs.google.com/spreadsheets/d/1oWS7cao-fgz2MKWtyr8h2dEL9unX__0bJrWKv6mQmM4/). The code that generates the YAMLs are in the [`./generators/google-sheets`](./generators/google-sheets) directory. The `regen-google-sheet` target in the Makefile can update the schema from Google Sheets automatically.
+The CRDC-H model is currently in development on [a Google Sheet](https://docs.google.com/spreadsheets/d/1oWS7cao-fgz2MKWtyr8h2dEL9unX__0bJrWKv6mQmM4/).
+This repository contains a program in the [`./generators/google-sheets`](./generators/google-sheets) directory,
+called `sheet2linkml`, which can read the Google Sheet using the Google Drive API and generate a LinkML representation
+of the model.
+
+In order to run this program, you will need to create and download Google Drive client credentials. First,
+[enable the Google Drive API](https://developers.google.com/drive/api/v3/enable-drive-api). After the API is enabled, 
+[create and download the client credentials](https://www.iperiusbackup.net/en/how-to-enable-google-drive-api-and-get-client-credentials/)
+from the [Google API Console](https://console.developers.google.com/). Save the file as `google_api_credentials.json` in
+the root directory of this project. [Detailed instructions and screenshots](https://pygsheets.readthedocs.io/en/stable/authorization.html)
```

Comment:
This works for me now in branch `merge-pipenv`. I still have the following files, relative to the project root:

- `./.env`
- `./sheets.googleapis.com-python.json`
- `./google_api_credentials.json`

I did find creating the Google API credentials and placing those files in the necessary locations confusing, _but I haven't done anything constructive about it yet!_

Reply:
I just tried setting this up from scratch, and I got it to work by putting my `google_api_credentials.json` file in the project root directory. It sounds like the same is true for Mark as well. Sujay: could you please confirm that you're definitely seeing this behavior on the latest version of this branch? If so, I'll try to replicate it again. You could also try to see if you can replicate this on GitHub Actions (#99).

Reply:
@gaurav: when I run `make generate-model` with the credentials in the root directory this is the log message I'm seeing.
```
➜  ccdhmodel git:(merge-pipenv) ✗ make generate-model
pipenv install
Installing dependencies from Pipfile.lock (6bfc04)...
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 2/2 — 00:00:09
pipenv install --dev
Installing dependencies from Pipfile.lock (6bfc04)...
Collecting mkdocs-material-extensions==1.0.37 — 00:00:08
  Using cached mkdocs_material_extensions-1.0.3-py3-none-any.whl (8.1 kB)
Installing collected packages: mkdocs-material-extensions
  Attempting uninstall: mkdocs-material-extensions
    Found existing installation: mkdocs-material-extensions 1.0.1
    Uninstalling mkdocs-material-extensions-1.0.1:
      Successfully uninstalled mkdocs-material-extensions-1.0.1
WARNING: Ignoring invalid distribution -kdocs (/Users/sujaypatil/.local/share/virtualenvs/ccdhmodel-3FU8GK2P/lib/python3.8/site-packages)
Successfully installed mkdocs-material-extensions-1.0.3
WARNING: Ignoring invalid distribution -ike (/Users/sujaypatil/.local/share/virtualenvs/ccdhmodel-3FU8GK2P/lib/python3.8/site-packages)
Collecting pathspec==0.9.0
  Using cached pathspec-0.9.0-py2.py3-none-any.whl (31 kB)
Installing collected packages: pathspec
Successfully installed pathspec-0.9.0
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 47/47 — 00:00:28
CDM_GOOGLE_SHEET_ID=1oWS7cao-fgz2MKWtyr8h2dEL9unX__0bJrWKv6mQmM4 PIPENV_IGNORE_INSTALLED=1 pipenv run python vendor/sheet2linkml/sheet2linkml.py --output model/schema/crdch_model.yaml
Loading .env environment variables...
Traceback (most recent call last):
  File "vendor/sheet2linkml/sheet2linkml.py", line 6, in <module>
    from sheet2linkml import cli
  File "/Users/sujaypatil/Desktop/ccdhmodel/vendor/sheet2linkml/sheet2linkml/cli.py", line 16, in <module>
    from sheet2linkml.source.gsheetmodel.gsheetmodel import GSheetModel
  File "/Users/sujaypatil/Desktop/ccdhmodel/vendor/sheet2linkml/sheet2linkml/source/gsheetmodel/gsheetmodel.py", line 2, in <module>
    from sheet2linkml.source.gsheetmodel.mappings import Mappings
  File "/Users/sujaypatil/Desktop/ccdhmodel/vendor/sheet2linkml/sheet2linkml/source/gsheetmodel/mappings.py", line 20, in <module>
    class Mappings:
  File "/Users/sujaypatil/Desktop/ccdhmodel/vendor/sheet2linkml/sheet2linkml/source/gsheetmodel/mappings.py", line 72, in Mappings
    ) -> list[Mapping]:
TypeError: 'type' object is not subscriptable
make: *** [generate-model] Error 1
```

So basically, you're right there's no problem with placing the credentials files in the root directory. I guess I was having problems with it when I was running it from a different branch.

But there seems to be a new error showing up like in the error message above.

Reply:
> So basically, you're right there's no problem with placing the credentials files in the root directory. I guess I was having problems with it when I was running it from a different branch.

Yay, good to know!

> But there seems to be a new error showing up like in the error message above.

That error is caused by generic types [not being supported](https://docs.python.org/3/library/typing.html#module-contents) before Python 3.9 -- if you re-run it with Python 3.9, it should work fine. Ideally, we would mark sheet2linkml as requiring Python 3.9, but we can't since it doesn't have its own setup.py file yet (#100). I've updated #100 so we remember to add that requirement to the setup.py file once we separate sheet2linkml into its own repo.

(The other possible solution would to be either remove the generic types or to replace them with the system being used before 3.9, but that system is going to be deprecated soon. So I think this is probably the best road forward. Note that ccdhmodel itself doesn't need Python 3.9 -- only sheet2linkml, which we're going to separate into its own repository pretty soon.)

Reply:
Ah, thanks for clarifying that @gaurav. Yup, `sheet2linkml` requiring Py 3.9+ is all the more reason why issue #100 is important.